File: trailing_zero/trailingzero.py
# find trailing zero in factorial of a number
# use optimized solution
def trailing_zero(num):
    count = 0
    i = 5
    while num // i >= 1:
        count += num // i
        i *= 5
    return count
# Example
print(trailing_zero(5))
print(trailing_zero(100))

File: trailing_zero/trailingzero.js
// program to find the trailing zeros in a factorial of a number
// naive approach
function trailingZero(num){
    var fact=1
    while(num>0){
        fact=fact*num
        num--
    }
    
    var count=0
    while(fact%10==0){
        count++
        fact=Math.floor(fact/10)
    }
    return count
}
console.log(trailingZero(5))
